core/templates/dev/head/services/AlertsService.ts

```diff
@@ -16,149 +16,161 @@
  * @fileoverview Factory for handling warnings and info messages.
  */
 
-angular.module('oppia').factory('AlertsService', ['$log', function($log) {
-  var AlertsService = {
-    /**
-     * Each element in each of the arrays here is an object with two keys:
-     *   - type:  a string specifying the type of message or warning.
-     *            Possible types - "warning", "info" or "success".
-     *   - content: a string containing the warning or message.
-     */
-
-    /**
-     * Array of "warning" messages.
-     */
-    warnings: [],
-    /**
-     * Array of "success" or "info" messages.
-     */
-    messages: [],
-
-    addWarning: null,
-    fatalWarning: null,
-    deleteWarning: null,
-    clearWarnings: null,
-    addMessage: null,
-    deleteMessage: null,
-    addInfoMessage: null,
-    addSuccessMessage: null,
-    clearMessages: null
-  };
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { LoggerService } from 'services/contextual/LoggerService';
+
+export interface Warning {
+  type: string;
+  content: string;
+}
+
+export interface Message {
+    type:string;
+    content:string;
+    timeout:number;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AlertsService {
+  /**
+   * Each element in each of the arrays here is an object with two keys:
+   *   - type:  a string specifying the type of message or warning.
+   *            Possible types - "warning", "info" or "success".
+   *   - content: a string containing the warning or message.
+   */
+
+  /**
+   * Array of "warning" messages.
+   */
+  warnings: Warning[] = new Array<Warning>();
+  /**
+   * Array of "success" or "info" messages.
+   */
+  messages: Message[] = new Array<Message>();
 
   // This is to prevent infinite loops.
-  var MAX_TOTAL_WARNINGS = 10;
-  var MAX_TOTAL_MESSAGES = 10;
+  MAX_TOTAL_WARNINGS: number = 10;
+  MAX_TOTAL_MESSAGES: number = 10;
+  log: LoggerService;
+
+  constructor(logger: LoggerService) {
+    this.log = logger;
+  }
 
   /**
    * Adds a warning message.
    * @param {string} warning - The warning message to display.
    */
-  AlertsService.addWarning = function(warning) {
-    $log.error(warning);
-    if (AlertsService.warnings.length >= MAX_TOTAL_WARNINGS) {
+  addWarning(warning: string): void {
+    this.log.warn(warning);
```

Comment:
I'm not sure why we are replacing error with warn? (Was it incorrect earlier?)

Reply:
I looked into your PR description and I have a few thoughts:
> We want the output to show as warning right? Since its present in addWarning function. I'm skeptical because we have $log.error in develop and here I'm adding log.warn.

I think in most of the places we have used the addWarning to report errors (like, errors from HTTP calls etc.). Considering that I think we should make it `error` as this will reflect back to the tests as an error and works as expected.

In addition to the above, I would say we can have an addError function maybe? (But this will need lots of work and investigation and I don't think that's should be a part of the migration or block the migration work. I would suggest you create an issue and add a TODO there.

Reply:
Replaced warn with error as you said. 

Why we need addError function? (So, that we restore our previous state, that is not logging into the terminal like this PR starts to do)